pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
+type MCDistributedPeers struct {
+	ClusterLabel []string `json:"cluster_label"`
+	ServerRoles  []string `json:"server_roles"`
+}
+
+//AutomateMCApplyChanges change the state of new indexers from "New" to "Configured" and add them in monitoring console asset table
+func (c *SplunkClient) AutomateMCApplyChanges(mock bool) error {
+	if mock {
+		return nil
+	}
+	var configuredPeers, indexerMemberList, licenseMasterMemberList string
+	apiResponseServerRoles, err := c.GetMonitoringconsoleServerRoles()
+	if err != nil {
+		return err
+	}
+
+	apiResponseMCDistributedPeers := struct {
+		Entry []struct {
+			Name    string             `json:"name"`
+			Content MCDistributedPeers `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/services/search/distributed/peers"
+	err = c.Get(path, &apiResponseMCDistributedPeers)
+	if err != nil {
+		return err
+	}
+
+	for _, e := range apiResponseMCDistributedPeers.Entry {
+		if configuredPeers == "" {
+			configuredPeers = e.Name
+		} else {
+			str := []string{configuredPeers, e.Name}
+			configuredPeers = strings.Join(str, ",")
+		}
+		for _, s := range e.Content.ServerRoles {
+			if s == "indexer" {
+				indexerMemberList = indexerMemberList + "&member=" + e.Name
+			}
+			if s == "license_master" {
+				licenseMasterMemberList = licenseMasterMemberList + "&member=" + e.Name
+			}
+		}
+	}
+
+	for _, e := range apiResponseServerRoles.ServerRoles {
+		if e == "indexer" {
+			indexerMemberList = "&member=localhost:localhost" + indexerMemberList
+		}
+		if e == "license_master" {
+			licenseMasterMemberList = licenseMasterMemberList + "&member=localhost:localhost"
+		}
+	}
+	reqBodyIndexer := indexerMemberList + "&default=true"
+	reqBodyLicenseMaster := licenseMasterMemberList + "&default=false"
+	err = c.UpdateDMCGroups("dmc_group_indexer", reqBodyIndexer)
+	if err != nil {
+		return err
+	}
+	err = c.UpdateDMCGroups("dmc_group_license_master", reqBodyLicenseMaster)
+	if err != nil {
+		return err
+	}
+
+	clusterRoleDict := make(map[string][]string)
+	//map of Name to Roles
+	for _, e := range apiResponseMCDistributedPeers.Entry {
+		for _, s := range e.Content.ClusterLabel {
+			clusterRoleDict[e.Name] = append(clusterRoleDict[e.Name], s)
+		}
+	}
+	//TODO: check different labels here
+	clusterRoleDictToDict := make(map[string][]string)
+	for key, value := range clusterRoleDict {
+		for _, val := range value {
+			clusterRoleDictToDict[val] = append(clusterRoleDictToDict[val], key)
+		}
+	}
+
+	clusterRoleDictToDictString := make(map[string]string)
+	for key, value := range clusterRoleDictToDict {
+		for _, val := range value {
+			clusterRoleDictToDictString[key] = clusterRoleDictToDictString[key] + "&member=" + val
+		}
+	}
+
+	for key, value := range clusterRoleDictToDictString {
+		if key == "" {
+			break
+		} else {
+			err = c.UpdateDMCClusteringLabelGroup(key, value)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	apiResponseMCAssetTableBuild, err := c.GetMonitoringconsoleAssetTable()
```

Comment:
Comments here would be nice to understand the functionality.

Reply:
we can take this offline, no need for too many comments in production code, I'll add comment on what the curl command request body requirements are. Example "curl -X POST -k -u admin:33Z9j0LZW3a6PoVTaJrncQvV https://127.0.0.1:8089/services/search/distributed/groups/dmc_group_indexer/edit -d member=192.168.31.25:8089 -d member=192.168.37.20:8089 -d member=192.168.47.138:8089 -d default=true"